openapi.yaml

```diff
@@ -0,0 +1,1791 @@
+          description: invalid Block identifier supplied
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /blocks/head:
+    get:
+      tags:
+      - blocks
+      summary: Get the most recently finalized block.
+      description: Returns the most recently finalized block
+      operationId: getHeadBlock
+      parameters:
+      - name: finalized
+        in: query
+        description: Boolean representing whether or not to get the finalized head.
+          If it is not set the value defaults to true. When set to false it will attempt
+          to get the newest known block, which may not be finalized. Replaces `/block`
+          from versions < v1.0.0.
+        required: false
+        schema:
+          type: boolean
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Block'
+  /node/network:
+    get:
+      tags:
+      - node
+      summary: Get network information of the node.
+      description: Returns network related information of the node.
+      operationId: getNodeNetworking
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NodeNetwork'
+  /node/transaction-pool:
+    get:
+      tags:
+      - node
+      summary: Get the transaction pool of the node.
+      description: Returns the extrinsics that the node knows of which have not 
+        been included in a block.
+      operationId: getNodeTransactionPool
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TransactionPool'
+  /node/version:
+    get:
+      tags:
+      - node
+      summary: Get version information of the node.
+      description: Returns versioning information of the node.
+      operationId: getNodeVersion
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NodeVersion'
+  /transaction:
+    post:
+      tags:
+      - transaction
+      summary: Submit a transaction to be included in the blockchain.
+      description: Accepts a valid signed extrinsic. Replaces `/tx` from versions
+        < v1.0.0.
+      operationId: submitTransaction
+      requestBody:
+        $ref: '#/components/requestBodies/Transaction'
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TransactionSuccess'
+        "400":
+          description: invalid transaction
```

Comment:
We should probably return different errors if Sidecar can't parse the tx vs. if the transaction pool returns an error.

Reply:
Agreed. So far I have not been specifying exact error messages, but instead status codes and the general them of them. Maybe @danforbes has some input as to how detailed the error response should be here?